Source/Scene/Cesium3DTileOptimizations.js

```diff
@@ -0,0 +1,110 @@
+                        var childOrientedBoundingBox = childBoundingVolume._orientedBoundingBox;
+
+                        // Compute the axis from the parent to the child.
+                        var axis = Cartesian3.subtract(childOrientedBoundingBox.center, orientedBoundingBox.center, scratchAxis);
+                        var axisLength = Cartesian3.magnitude(axis);
+                        Cartesian3.divideByScalar(axis, axisLength, axis);
+
+                        // Project the bounding box of the parent onto the axis. Because the axis is a ray from the parent to the child,
+                        // the projection parameterized along the ray will be (+/- proj1).
+                        var proj1 = Math.abs(orientedBoundingBox.halfAxes[0] * axis.x) +
+                                    Math.abs(orientedBoundingBox.halfAxes[1] * axis.y) +
+                                    Math.abs(orientedBoundingBox.halfAxes[2] * axis.z) +
+                                    Math.abs(orientedBoundingBox.halfAxes[3] * axis.x) +
+                                    Math.abs(orientedBoundingBox.halfAxes[4] * axis.y) +
+                                    Math.abs(orientedBoundingBox.halfAxes[5] * axis.z) +
+                                    Math.abs(orientedBoundingBox.halfAxes[6] * axis.x) +
+                                    Math.abs(orientedBoundingBox.halfAxes[7] * axis.y) + 
+                                    Math.abs(orientedBoundingBox.halfAxes[8] * axis.z);
+                        
+                        // Project the bounding box of the child onto the axis. Because the axis is a ray from the parent to the child,
+                        // the projection parameterized along the ray will be (+/- proj2) + axis.length.
+                        var proj2 = Math.abs(childOrientedBoundingBox.halfAxes[0] * axis.x) +
+                                    Math.abs(childOrientedBoundingBox.halfAxes[1] * axis.y) +
+                                    Math.abs(childOrientedBoundingBox.halfAxes[2] * axis.z) +
+                                    Math.abs(childOrientedBoundingBox.halfAxes[3] * axis.x) +
+                                    Math.abs(childOrientedBoundingBox.halfAxes[4] * axis.y) +
+                                    Math.abs(childOrientedBoundingBox.halfAxes[5] * axis.z) +
+                                    Math.abs(childOrientedBoundingBox.halfAxes[6] * axis.x) +
+                                    Math.abs(childOrientedBoundingBox.halfAxes[7] * axis.y) +
+                                    Math.abs(childOrientedBoundingBox.halfAxes[8] * axis.z);
+                        
+                        // If the child extends the parent's bounds, the optimization is not valid and we skip it.
+                        if (proj1 <= proj2 + axisLength) {
+                            tile._optimChildrenWithinParent = Cesium3DTileOptimizationHint.SKIP_OPTIMIZATION;
+                            break;
+                        }
```

Comment:
Perhaps, but it's not a totally accurate OrientedBoundingBox.contains(OrientedBoundingBox) check. It becomes less useful if the parent and child don't have similar axes. In this case it's not a problem for two reasons. Most of the time the chidlren do have the same axes as their parent. Secondly, if it is wrong, we will just skip the optimization and not produce visual artifacts..